### PR TITLE
Validation list

### PR DESCRIPTION
**Pull request by goivvy**
All available form validation rules sorted alphabetically.

## Purpose of this pull request

This PR adds a form validation rule list.

Fixes https://github.com/magento/devdocs/issues/8951

whatsnew
Added a remarkable [Validation rule list](https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/validations/rule-list.html) to the Frontend Developer's Guide.

### CODE REVIEW

**Comment by dobooth** [comment]
Wow! Amazing effort here!

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 1]
```diff
@@ -0,0 +1,1664 @@
...
+   <div class="control">
+     <input name="field-4" id="field-5" title="Field 5" type="text" data-validate="{'validate-xml-identifier':true}"/>
+   </div>
+...
+</form>
+```
+
+### validate-zero-or-greater
+
+It checks for a number 0 or greater in this field.
```
Comment (BarnyShergold) [inline src/guides/v2.3/frontend-dev-guide/validations/rule-list.md:1588]:
```suggestion
It checks that a number is zero or greater in this field.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
This is a brilliant piece of work!

Please correct as many changes by accepting the suggestion on the repo itself. Please DO NOT correct locally and then push a new commit - this is because it will be really difficult to check all the updates requested have been done as there are over 60.

You also need to do the following -
1. Add this new file to the TOC file - src/_data/toc - around line 189 - currently there is no way to get to your new page
2. Add a SYMLINK to the new file in the src/guides/2.4 area

Please also fill in the PR description as requested by the WIKI - this is very important.

Well Done!

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 2]
```diff
@@ -0,0 +1,1664 @@
...
+  <input name="example" id="example" title="Example" value="" type="text" data-validate="{'max-words':4}"/>
+...
+</form>
+```
+
+Here, it accepts no more than 4 words.
+
+### min-words
+
+It checks for more than predefined number of words. Min number of words should be set as a parameter.
```
[the remark this answers is not in the document]
Comment (goivvy) [reply src/guides/v2.3/frontend-dev-guide/validations/rule-list.md]: fixed.

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 3]
```diff
@@ -0,0 +1,1664 @@
...
+<form class="form" id="my-form" method="post" data-mage-init='{"validation": {}}'>
+...
+ <input name="field-4" id="field-3" title="Field 3" type="text" data-validate="{'time':true}"/>
+...
+</form>
+```
+
+### time12h
+
+It checks for a valid time between 00:00 am and 12:00 pm.
```
[the remark this answers is not in the document]
Comment (goivvy) [reply src/guides/v2.3/frontend-dev-guide/validations/rule-list.md]: fixed.

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 4]
```diff
@@ -0,0 +1,1664 @@
...
+```html
+<form class="form" id="my-form" method="post" data-mage-init='{"validation": {}}'>
+...
+  <input name="field-4" id="field-3" title="Field 3" type="text" data-validate="{'validate-cc-type':'#field-5'}"/>
+  <input name="field-5" id="field-5" type="text" value="AE"/>
+...
+</form>
+```
+
+Field `field-5` holds a credit card type, `AE` stands for American Express.
```
[the remark this answers is not in the document]
Comment (goivvy) [reply src/guides/v2.3/frontend-dev-guide/validations/rule-list.md]: fixed.

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 5]
```diff
@@ -0,0 +1,1664 @@
...
+   <div class="control">
+     <input name="field-4" id="field-5" title="Field 5" type="text" data-validate="{'validate-ssn':true}"/>
+   </div>
+...
+</form>
+```
+
+### validate-state
+
+It checks for a valid state.
```
[the remark this answers is not in the document]
Comment (goivvy) [reply src/guides/v2.3/frontend-dev-guide/validations/rule-list.md]: fixed.

**Comment by goivvy** [comment]
@BarnyShergold done, please review.

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 6]
```diff
@@ -0,0 +1,1664 @@
...
+```html
+<form class="form" id="my-form" method="post" data-mage-init='{"validation": {}}'>
+...
+  <input name="field-4" id="field-3" title="Field 3" type="text" data-validate="{'validate-cc-type':'#field-5'}"/>
+  <input name="field-5" id="field-5" type="text" value="AE"/>
+...
+</form>
+```
+
+Field `field-5` holds a credit card type, possible values are: `SO`, `SM`, `VI`, `MC`, `AE`, `DI`, `JCB`, `DN`, `UN`, `MI`, `MD`.
```
Comment (BarnyShergold) [inline src/guides/v2.3/frontend-dev-guide/validations/rule-list.md:781]: As there are so many and some are not obvious - I'd recommend turning this list into a small table and defining what each of the 2 letter codes are.

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md [hunk 7]
```diff
@@ -0,0 +1,1664 @@
...
+<form class="form" id="my-form" method="post" data-mage-init='{"validation": {}}'>
+...
+ <input name="field-4" id="field-3" title="Field 3" type="text" data-validate="{'time':true}"/>
+...
+</form>
+```
+
+### time12h
+
+It checks for a valid time between 00:00 am and 12:00 pm. `12:01 pm` and `11:59 pm` value are valid.
```
Comment (BarnyShergold) [inline src/guides/v2.3/frontend-dev-guide/validations/rule-list.md:610]:
```suggestion
It checks for a valid time between 00:00 am and 12:00 pm. `12:01 pm` and `11:59 pm` values are also valid.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Awesome turnround speed!
I've made one correction and a suggestion to improve clarity and then it's good to go to
@dobooth - might be worth a second reviewer as there is so much - I may have missed something.

But this is an awesome addition.

File: src/guides/v2.3/frontend-dev-guide/validations/rule-list.md (hunk 6 above)
Comment (goivvy) [reply src/guides/v2.3/frontend-dev-guide/validations/rule-list.md]: Fixed.

**Comment by goivvy** [comment]
@BarnyShergold 
fixed.

**Review by dobooth — APPROVED** [review]

**Comment by m2-github-services** [comment]
@dobooth the branch with the code is successfully imported into the `magento-devdocs/devdocs` repository. Branch name: `imported-goivvy-devdocs-9356`.

**Pull request merged**